### PR TITLE
docs: rewrite docs for type inference

### PR DESCRIPTION
**Pull request by juliusmarminge**
Referencing this comment on a pull request on trpc/zart:
https://github.com/trpc/zart/pull/24#discussion_r926082681

We should probably rewrite the inferance-helper section as well. I began drafting this out but quickly realized I didn't have enough knowledge of how this new way worked and how to document this so I think I'l leave it to someone else to pick up on, or if you can point some stuff out that I can continue upon.

### CODE REVIEW

File: www/docs/server/infer-types.md [hunk 1]
```diff
@@ -15,81 +15,37 @@ It is often useful to wrap functionality of your `@trpc/client` or `@trpc/react`
 - `inferProcedureInput<TProcedure>`
 - `inferSubscriptionOutput<TRouter, TPath>`
```
Comment (sachinraja) [inline www/docs/server/infer-types.md:16]: I think this should be removed

Comment (KATT) [reply www/docs/server/infer-types.md:16]: Agreed!

Comment (juliusmarminge) [reply www/docs/server/infer-types.md:16]: Ok. Wasn't sure about the API of the subscription so I just left it there for now. Can remove it. Anything else we should add?

**Comment by KATT** [comment]
shippp iiit

**Pull request merged**